### PR TITLE
Test LIKE with dynamic pattern 

### PR DESCRIPTION
**Pull request by findepi**
LIKE pattern doesn't have to be static. Add test cases with dynamic LIKE
to prevent potential regressions. The existing test cases https://github.com/apache/datafusion/blob/8aafa5498024e45b58ad2068a80cf5942babe55b/datafusion/sqllogictest/test_files/select.slt#L670-L685 are not exhaustive yet.

### CODE REVIEW

**Comment by jatin510** [comment]
Hi @findepi,

I was reviewing the PR, and I have a few questions:

- Could you explain what issues these test cases are meant to solve?
- What does "dynamic pattern" refer to in this context?
- How are these dynamic patterns generated for testing?

I'm still learning about how the LIKE operator works with different patterns, and I’d appreciate any insights you can share to help me understand the purpose and impact of these tests. 
Thanks so much for your time and effort! 😀

**Comment by findepi** [comment]
@jatin510 thanks for your review.
by "dynamic pattern" i mean opposite of static pattern, ie literal. So `LIKE another_column`

File: datafusion/sqllogictest/test_files/string/string.slt [hunk 1]
```diff
@@ -48,13 +48,115 @@ from test_basic_operator;
...
+under_score un iść core false false false false
+percent pan Tadeusz ma iść w kąt false false false false
 NULL NULL NULL NULL NULL NULL
 
 #
 # common test for string-like functions and operators
 #
 include ./string_query.slt.part
 
+# TODO support all String types in sql_like_to_expr and move this test to `string_query.slt.part`
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/string/string.slt:60]: Is this the same as https://github.com/apache/datafusion/issues/12637? 

If so perhaps we can add a reference to the ticket as a comment

If not, maybe we can create another issue -- is the issue it doesn't work in StringView? / Dictionary?

**Review by alamb — APPROVED** [review]
Thank you @findepi  -- this seems like an improvement to me. I have just one question about the queries that can't run in `string_query.slt.part`

**Comment by alamb** [comment]
Thanks again @findepi

**Pull request merged**